### PR TITLE
[tests-only] [full-ci] added api step to assign tag on resources

### PR DESCRIPTION
**Pull request by PrajwolAmatya**
## Description
This PR adds step implementation of assigning tag on resources using API.
```
And "Alice" adds the following tags for the following resources using API
      | resource        | tags      |
      | fileToShare.txt | alice tag |
```

## Related Issue

- https://github.com/owncloud/web/issues/9157

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/e2e/cucumber/steps/api.ts [hunk 1]
```diff
@@ -298,3 +298,13 @@ Given(
     }
   }
 )
+
+Given(
+  '{string} adds the following tags for the following resources using API',
+  async function (this: World, stepUser: string, stepTable: DataTable): Promise<void> {
+    const user = this.usersEnvironment.getUser({ key: stepUser })
+    for (const data of stepTable.hashes()) {
+      await api.dav.addTagToResource({ user, data: data })
```
Comment (saw-jan) [inline tests/e2e/cucumber/steps/api.ts:307]:
```suggestion
      await api.dav.addTagToResource({ user, data })
```

File: tests/e2e/support/api/davSpaces/spaces.ts [hunk 2]
```diff
@@ -188,16 +191,19 @@ export const getDataOfFileInsideSpace = async ({
...
   pathToFileName: string
+  spaceType: string
   spaceName: string
 }): Promise<string> => {
   const fileDataResponse = await getDataOfFileInsideSpace({
     user,
     pathToFileName,
-    spaceName
+    spaceType: spaceType,
+    spaceName: spaceName
```
Comment (saw-jan) [inline tests/e2e/support/api/davSpaces/spaces.ts:206]:
```suggestion
    spaceType,
    spaceName
```

File: tests/e2e/support/api/davSpaces/spaces.ts [hunk 3]
```diff
@@ -219,3 +225,14 @@ export const addMembersToTheProjectSpace = async ({
...
+
+export const addTagToResource = async ({ user, data }: { user: User; data: object }) => {
+  const resourceId = await getIdOfFileInsideSpace({
+    user,
+    pathToFileName: data['resource'],
+    spaceType: 'personal',
+    spaceName: user.displayName
+  })
+  const tagNameArray = data['tags'].split(',')
+  await createTagsForResource({ user, resourceId, tagNames: tagNameArray })
```
Comment (saw-jan) [inline tests/e2e/support/api/davSpaces/spaces.ts:237]:
```suggestion
  const tags = data['tags'].split(',')
  await createTagsForResource({ user, resourceId, tagNames: tags })
```

File: tests/e2e/support/api/graph/utils.ts [hunk 4]
```diff
@@ -22,3 +22,24 @@ export const getApplicationEntity = async (admin: User): Promise<ApplicationEnti
...
+  tagNames
+}: {
+  user: User
+  resourceId: string
+  tagNames: Array<string>
+}) => {
+  const body = {
+    resourceId: resourceId,
+    tags: tagNames
+  }
```
Comment (saw-jan) [inline tests/e2e/support/api/graph/utils.ts:38]:
```suggestion
  resourceId,
  tagNames
}: {
  user: User
  resourceId: string
  tags: Array<string>
}) => {
  const body = {
    resourceId,
    tags
  }
```

File: tests/e2e/cucumber/steps/api.ts (hunk 1 above)
Comment (saw-jan) [inline tests/e2e/cucumber/steps/api.ts:307]:
```suggestion
      await api.dav.addTagToResource({ user, resource: data.resource, tags: data.tags })
```

File: tests/e2e/support/api/graph/utils.ts [hunk 5]
```diff
@@ -22,3 +22,24 @@ export const getApplicationEntity = async (admin: User): Promise<ApplicationEnti
...
+  const body = {
+    resourceId,
+    tags
+  }
+  await request({
+    method: 'PUT',
+    path: join('graph', 'v1.0', 'extensions', 'org.libregraph', 'tags'),
+    body: JSON.stringify(body),
+    user: user
+  })
```
Comment (SagarGi) [inline tests/e2e/support/api/graph/utils.ts:44]:
```suggestion
  const response = await request({
    method: 'PUT',
    path: join('graph', 'v1.0', 'extensions', 'org.libregraph', 'tags'),
    body: JSON.stringify(body),
    user: user
  })
  checkResponseStatus(response, 'Failed while  creating tags for a resource.'
```

File: tests/e2e/support/api/davSpaces/spaces.ts [hunk 6]
```diff
@@ -219,3 +225,24 @@ export const addMembersToTheProjectSpace = async ({
...
+}): Promise<void> => {
+  const resourceId = await getIdOfFileInsideSpace({
+    user,
+    pathToFileName: resource,
+    spaceType: 'personal',
+    spaceName: user.displayName
+  })
+  const tagNames = tags.split(',').map((tag) => {
+    return tag.trim()
+  })
```
Comment (saw-jan) [inline tests/e2e/support/api/davSpaces/spaces.ts:246]:
```suggestion
  const tagNames = tags.split(',').map((tag) => tag.trim())
```

**Review by saw-jan — COMMENTED** [review]
Others LGTM

**Review by SagarGi — COMMENTED** [review]
LGTM :+1:

**Review by saw-jan — APPROVED** [review]
:+1:

**Review by ScharfViktor — APPROVED** [review]

**Pull request merged**